geode-core/src/main/java/org/apache/geode/internal/cache/AbstractRegion.java

```diff
@@ -1867,6 +1874,17 @@ public void incRecentlyUsed() {
     // nothing
   }
 
+  /**
+   * Only subclasses of {@code AbstractRegion} should use this supplier to acquire the
+   * {@code StatisticsClock}.
+   *
+   * <p>
+   * Please do not use this accessor from any class other than a Region.
+   */
```

Comment:
What's the danger? (I don't necessarily need an explanation in the comment, but I am puzzled about this admonition.)

Reply:
Not really any danger. I just really wanted every class to have its own `statisticsClock` field. So this is the field for `AbstractRegion` and I wanted to use an accessor from the subclasses instead of directly accessing the field. But I didn't want people to see this and think `oh, I'll just call region.getStatisticsClock() instead of injecting it into the constructor of this new/old class`. Maybe I should instead introduce a new annotation `@VisibleForSubclasses` or something like that?

Reply:
Or maybe `@ProtectedForSubclass`